byway/src/main/webapp/destinations.js

```diff
@@ -33,8 +33,9 @@ function initializeDestinationsPage(){
   }); 
 }
 
-const urlParams = new URLSearchParams(window.location.search);
-const tripKey = urlParams.get('tripKey');
+const tripKey = getTripKeyFromUrl();
+
+let service;
```

Comment:
Nit: give this a slightly more descriptive name like `PlacesService`

Reply:
done